base/reflectx/pointers.go

```diff
@@ -88,16 +88,22 @@ func UnderlyingPointer(v reflect.Value) reflect.Value {
 	return OnePointerValue(uv)
 }
 
-// NewFrom returns a value that is guaranteed to be a pointer to the [Underlying] version of
-// the given value. If that value is not addressable, it makes a new fake pointer that points
-// to a copy of the value, not the actual value. This should only be used you do not need the
-// pointer to actually point to the original value.
-func NewFrom(v reflect.Value) reflect.Value {
-	u := Underlying(v)
-	if u.CanAddr() {
-		return u.Addr()
+// NonPointerNew has the same overall behavior as [reflect.New] except that
+// it traverses through any pointers such that a new zero non-pointer value
+// will be created in the end, so any pointers in the original type will not
+// be nil. For example, in pseudo-code, NonPointerNew(**int) will return
+// &(&(&(0))).
+func NonPointerNew(typ reflect.Type) reflect.Value {
```

Comment:
NonNilPointerNew might be a better name?  It is not getting rid of the pointers, right?  just making them non-nil.

Reply:
`NonNilPointerNew` still seems like a very awkward name; we should be able to come up with something better. I thought about `DeepNew` before (like `DeepEqual`), but that is somewhat unclear as well.

Reply:
How about `SafePointerNew` or just `SafeNew` ?

Reply:
Going with `NonNilNew`, which is both concise and clear.